src/test/java/org/broadinstitute/consent/http/resources/InstitutionResourceTest.java

```diff
@@ -0,0 +1,160 @@
+    Institution mockInstitution = institutionSetup();
+    institutions.add(mockInstitution);
+    when(userService.findUserByEmail(anyString())).thenReturn(researcherUser);
+    when(institutionService.findAllInstitutions()).thenReturn(institutions);
+    initResource();
+    Response researcherResponse = resource.getInstitutions(authUser);
+    String json = researcherResponse.getEntity().toString();
+    assertEquals(200, researcherResponse.getStatus());
+    assertNotNull(json);
+  }
+
+  @Test
+  public void getInstitutionAdmin() {
+    Institution mockInstitution = institutionSetup();
+    when(userService.findUserByEmail(anyString())).thenReturn(adminUser);
+    when(institutionService.findInstitutionById(anyInt())).thenReturn(mockInstitution);
+    String id = "1";
+    Response adminResponse = resource.getInstitution(authUser, id);
+    String json = adminResponse.getEntity().toString();
+    assertEquals(adminResponse.getStatus(), 200);
+    assertNotNull(json);
+  }
+
+  @Test
+  public void getInstitutionNonAdmin() {
+    Institution mockInstitution = institutionSetup();
+    when(userService.findUserByEmail(anyString())).thenReturn(researcherUser);
+    when(institutionService.findInstitutionById(anyInt())).thenReturn(mockInstitution);
+    String id = "1";
+    Response researcherResponse = resource.getInstitution(authUser, id);
+    String json = researcherResponse.getEntity().toString();
+    assertEquals(200, researcherResponse.getStatus());
+    assertNotNull(json);
+  }
+
+  @Test
+  public void testCreateInstitution() {
+    Institution mockInstitution = institutionSetup();
+    when(userService.findUserByEmail(anyString())).thenReturn(adminUser);
+    when(institutionService.createInstitution(any(), anyInt())).thenReturn(mockInstitution);
+    String requestJson = new Gson().toJson(mockInstitution, Institution.class);
+    Response response = resource.createInstitution(authUser, requestJson);
+    String json = response.getEntity().toString();
+    assertEquals(200, response.getStatus());
+    assertNotNull(json);
+  }
+
+  @Test
+  public void testUpdateInstitution() {
+    Institution mockInstitution = institutionSetup();
+    when(userService.findUserByEmail(anyString())).thenReturn(adminUser);
+    when(institutionService.updateInstitutionById(any(), anyInt(), anyInt())).thenReturn(mockInstitution);
+    String paramId = "1";
+    Response response = resource.updateInstitution(authUser, paramId, new Gson().toJson(mockInstitution));
+    assertEquals(200, response.getStatus());
+    assertNotNull(response.getEntity().toString());
+  }
+
+  @Test
+  public void testDeleteInstitution() { 
+    when(userService.findUserByEmail(anyString())).thenReturn(adminUser);
+    when(institutionService.updateInstitutionById(any(), anyInt(), anyInt())).thenReturn(null);
+    String paramId = "1";
+    Response response = resource.deleteInstitution(authUser, paramId);
+    assertEquals(200, response.getStatus());
+  }  
+}
```

Comment:
Other tests to consider are the 404's and making sure that response codes are appropriate.